weather/module.py

```diff
@@ -1,23 +1,155 @@
 from typing import Optional
+import aiohttp
 
+import nextcord
 from nextcord.ext import commands
 
 import pie.database.config
-from pie import check, i18n, logger
+from pie import check, i18n, logger, utils
+from pie.i18n.database import GuildLanguage, MemberLanguage
 
 from .database import Place
 
 _ = i18n.Translator("modules/fun").translate
 guild_log = logger.Guild.logger()
 config = pie.database.config.Config.get()
 
+# number of days to get forecast for (including current day, max is 3)
+NUM_OF_FORECAST_DAYS = 3
+# dict for getting the data from json easier (when you don't wan't some phase of day comment it)
+DAY_PHASES = {
+    "Morning": 2,
+    "Day": 4,
+    # "Evening": 6,
+    "Night": 7,
+}
+
 
 class Weather(commands.Cog):
     """Weather and forecast"""
 
-    def __init__(self, bot):
+    def __init__(self, bot: commands.Bot):
         self.bot = bot
 
+    def _get_useful_data(
+        self, json: dict, ctx: commands.Context, lang_preference: str
+    ) -> list[dict]:
+        """
+        example json: https://wttr.in/praha?lang=sk&format=j1
+        get useful data from json as list of individual days
+        """
+
+        # get individual days to extract data
+        weather = []
+        nearest_place = json["nearest_area"][0]["areaName"][0]["value"]
+        lang_preference = f"lang_{lang_preference}"
+        for i in range(NUM_OF_FORECAST_DAYS):
+            day = json["weather"][i]
+            day_dict = {
+                "date": day["date"],
+                "nearest_place": nearest_place,
+            }
+            day = day["hourly"]
+            for day_phase, hour in DAY_PHASES.items():
+                if lang_preference != "lang_en":
+                    day_dict.update(
+                        {
+                            _(ctx, f"{day_phase}"): {
+                                "state": day[hour][lang_preference][0]["value"],
+                                "temp": day[hour]["tempC"],
+                                "feels_like": day[hour]["FeelsLikeC"],
+                                "wind_speed": day[hour]["windspeedKmph"],
+                                "rain_chance": day[hour]["chanceofrain"],
+                            }
+                        }
+                    )
+                else:
+                    day_dict.update(
+                        {
+                            _(ctx, f"{day_phase}"): {
+                                "state": day[hour]["weatherDesc"][0]["value"],
+                                "temp": day[hour]["tempC"],
+                                "feels_like": day[hour]["FeelsLikeC"],
+                                "wind_speed": day[hour]["windspeedKmph"],
+                                "rain_chance": day[hour]["chanceofrain"],
+                            }
+                        }
+                    )
```

Comment:
1/ Don't they to the same thing?
2/ `f"{day_phase}"` is the same as just `day_phase`, or is there a reason for writing it like that?

Reply:
They are very similiar but they don't do the same thing. But I think it could be simplified.

Reply:
Also, row 57 and 69 are gonna cause trouble. You can't put formatted string into _(ctx, ...) as parameter.

Reply:
Also as I mentioned at Discord, there's only one difference base on lang_preference - value for state. You should assign that value into some variable in if/else block and whole day_dict.update(...) use only once with this variable.

Reply:
Yes, that is what I meant by "I think it could be simplified" in my comment above.